source/Camera/Basic/C++Halcon/CaptureViaGenICam/CaptureViaGenICam.cpp

```diff
@@ -86,6 +82,9 @@ int main()
 
 		setColorsInObjectModel3D(objectModel3D, RGB);
 
+		std::cout << "Removing NaN points" << std::endl; // This is useful for processing the point cloud later (e.g. surface-based matching)
+		objectModel3D = objectModel3D.SelectPointsObjectModel3d("point_coord_z", 0.0001, 10000.0);
```

Comment:
Weren't you going to try to do xyz mappings beforehand if possible?

Reply:
It was not possible, then I could not add color information. Asked MVTec about this, still waiting for their reply

Reply:
But if you don't need color then it is possible?

Reply:
We do already have xyz_mapping, it comes when we grab data and construct the objectModel3d. What I don't know how to do is remove all NaN points right after or before we create objectModel3d

Reply:
We have xyz_mappings but it includes all the points, including NaNs=zeros. It treats them as valid points, which is wrong.
I think that we should re-apply correct xyz_mappings somehow after the ObjectModel3D is created. We can probably get it from either of X Y Z images.

If the colour is indeed impossible to add afterwards, then we should consider two routes: color -> as it is now (removing points afterwards), and no_color -> with xyz_mappings modification (if it is faster)

Reply:
done. XYZ mapping is done before I crea<et ObjectModel3D now.

Reply:
Is the method with 'Threshold' and 'ReduceDomain' better for some reason than the one with 'SelectPoints'?

Reply:
@chrisasc

Reply:
I'm pretty sure it is because of two reasons:
1) We now only create one objectmodel3d instead of two. This might save some time. 
2) We now only remove NaN points instead of thresholding on z-values. This preserves all valid points, while z-thresholding might remove valid points. MVTec also stated that this is more "correct" than just thresholding z-values.